app/services/usps_exporter.rb

```diff
@@ -0,0 +1,46 @@
+class UspsExporter
+  OTP_MAX_VALID_DAYS = 30
+
+  def initialize(csv_file_path)
+    @csv_file_path = csv_file_path
+  end
+
+  def run
+    CSV.open(csv_file_path, 'wb', col_sep: '|') do |csv|
+      entries.each do |entry|
+        csv << make_entry_row(entry)
+      end
+    end
+    clear_entries
+  end
+
+  private
+
+  attr_reader :csv_file_path
+
+  def entries
+    UspsConfirmation.all.map(&:decrypted_entry)
+  end
+
+  def clear_entries
+    UspsConfirmation.destroy_all
```

Comment:
My expectation is that this job is run once daily.

Yes, I was going to approach the SFTP upload as a separate step, just for separation of concerns. My assumption is that the file creation was w/in our control, but the upload has external dependencies, so was going to tackle them separately. E.g. the file could be written to a specified directory, and then the SFTP task could try (and re-try) till the file is successfully confirmed upload, then remove the file. Same idea (don't remove the data we are sure the vendor has it).